x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/common/time/TimeUtils.java

```diff
@@ -46,6 +46,37 @@ public static Instant parseTimeFieldToInstant(XContentParser parser, String fiel
             "unexpected token [" + parser.currentToken() + "] for [" + fieldName + "]");
     }
 
+    /**
+     * Safely parses a string epoch representation to a Long
+     *
+     * Commonly this function is used for parsing Date fields from doc values
+     * requested with the format "epoch_millis".
+     *
+     * Since nanosecond support was added epoch_millis timestamps may have a fractional component.
+     * We discard this, taking just whole milliseconds.  Arguably it would be better to retain the
+     * precision here and let the downstream component decide whether it wants the accuracy, but
+     * that makes it hard to pass around the value as a number.  The double type doesn't have
+     * enough digits of accuracy, and obviously long cannot store the fraction.  BigDecimal would
+     * work, but that isn't supported by the JSON parser if the number gets round-tripped through
+     * JSON.  So String is really the only format that could be used, but the consumers of time
+     * are expecting a number.
+     *
+     * @param epoch The
+     * @return The epoch value.
+     */
+    public static Long parseToEpochMs(String epoch) {
```

Comment:
Would it be better to return `long`?  In the case where the caller wants `Long` it can be boxed just as efficiently on the caller's side.  And if the caller wants `long` then there's no boxing and unboxing.